lib/timerage/time_interval.rb

```diff
@@ -135,7 +135,7 @@ def rangeish?(an_obj)
     end
 
     def time_enumerator(step)
-      next_offset = 0.seconds
+      next_offset = (self.begin + step).is_a?(Date) ? 0.days : 0.seconds
```

Comment:
what if we just make the initial offset `0`? i think that will make types come out right and it will simplify this line 
```suggestion
      next_offset = 0
```

Reply:
`0` will cause the opposite issue to occur. 

When given a date range and you step through in hour slices , the first object will be a Date object and the rest Time objects.

Reply:
oh, i see! i still hate the type check. How about multiply step by 0 for the first offset

```suggestion
      next_offset = step * 0
```

```rb
Date.today + (1.hour * 0)
# => 2023-10-16 00:00:00 -0600
Date.today + (1.day * 0)
# => Mon, 16 Oct 2023
```

Reply:
That is a very elegant solution ✨